openmc/source.py

```diff
@@ -379,6 +393,139 @@ def from_xml_element(cls, elem: ET.Element, meshes=None) -> 'openmc.SourceBase':
 
         return source
 
+class MeshSource(SourceBase):
+    """A mesh-based source in which random positions are uniformly sampled
+    within mesh elements and each element can have independent angle, energy,
+    and time distributions. The element sampled is chosen based on the relative
+    strengths of the sources applied to the elements. The strength of the mesh
+    source as a whole is the sum of all source strengths applied to the elements.
+
+    Parameters
+    ----------
+    mesh : openmc.MeshBase
+        The mesh on which source sites will be generated.
+    sources : iterable of openmc.SourceBase
+        Sources for each element in the mesh. If spatial distributions are set
+        on any of the source objects, they will be ignored during source site
```

Comment:
Would it be worth raising an error or a warning is a spatial distribution is set on the source